sound/soc/sof/intel/mtl.c

```diff
@@ -612,6 +613,33 @@ static int mtl_dsp_disable_interrupts(struct snd_sof_dev *sdev)
 	return mtl_disable_interrupts(sdev);
 }
 
+static int mtl_dsp_update_stream_position(struct snd_sof_dev *sdev,
+					  struct snd_soc_component *component,
+					  struct snd_pcm_substream *substream)
+{
+	struct snd_soc_pcm_runtime *rtd = asoc_substream_to_rtd(substream);
+	struct hdac_stream *hstream = substream->runtime->private_data;
+	struct snd_soc_component *scomp = sdev->component;
+	struct sof_ipc_stream_posn *posn;
+	struct snd_sof_pcm *spcm;
+	u32 host;
+	u32 llp_l, llp_u;
+
+	spcm = snd_sof_find_spcm_dai(scomp, rtd);
+	posn = &spcm->stream[substream->stream].posn;
+
+	host = substream->runtime->status->hw_ptr;
+	posn->host_posn = bytes_to_frames(substream->runtime, host);
```

Comment:
Updating the host_posn can be done with this exact code up, in the generic code, before calling in to `update_stream_position()`, no?

Then the `update_stream_position()` can be renamed to `get_stream_position()`, return the dai_posn, so it can be documented that it is retrieving the dai position for delay calculation?

`update_stream_position()` is a but broad and can be seen as it is moving the stream position, updates it somewhere (where?) and so on.

Reply:
thanks, very reasonable.